src/librustc_trans/mir/rvalue.rs

```diff
@@ -429,6 +429,24 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
                 })
             }
 
+            mir::Rvalue::Discriminant(ref lvalue) => {
+                let discr_lvalue = self.trans_lvalue(&bcx, lvalue);
+                let enum_ty = discr_lvalue.ty.to_ty(bcx.tcx());
+                let discr_ty = rvalue.ty(&*self.mir, bcx.tcx()).unwrap();
+                let discr_type = type_of::immediate_type_of(bcx.ccx, discr_ty);
+                // FIXME: inline this
```

Comment:
This is what I’d expect would happen to this function once we finally remove the old trans. Since now MIR can express the `Discriminant` rvalue, it is possible to port all the old trans bits which deal with discriminants (e.g. `discriminant` intrinsic) to the MIR trans.

That being said, this FIXME should not exist currently, as there a fair amount of work involved in making this happen still.